src/Microsoft.CodeQuality.Analyzers/VisualBasic/Maintainability/BasicRemoveUnusedLocals.Fixer.vb

```diff
@@ -19,5 +22,50 @@ Namespace Microsoft.CodeQuality.VisualBasic.Analyzers.Maintainability
                 Return ImmutableArray.Create(BasicRemoveUnusedLocalsAnalyzer.RuleId)
             End Get
         End Property
+
+        Public Sub New()
+            MyBase.New(New BasicNodesProvider())
+        End Sub
+
+        Private Class BasicNodesProvider
+            Inherits NodesProvider
+
+            Public Overrides Function GetNodeToRemoveOrReplace(node As SyntaxNode) As SyntaxNode
+                node = node.Parent
+                If (node.Kind() = SyntaxKind.SimpleAssignmentStatement) Then
+                    Return node
+                End If
+                Return Nothing
+            End Function
+
+            Public Overrides Sub RemoveAllUnusedLocalDeclarations(nodesToRemove As HashSet(Of SyntaxNode))
+                Dim candidateLocalDeclarationsToRemove = New HashSet(Of LocalDeclarationStatementSyntax)()
+                For Each variableDeclarator In nodesToRemove.OfType(Of VariableDeclaratorSyntax)()
+                    Dim localDeclaration = DirectCast(variableDeclarator.Parent.Parent, LocalDeclarationStatementSyntax)
+                    candidateLocalDeclarationsToRemove.Add(localDeclaration)
+                Next
+
+                For Each candidate In candidateLocalDeclarationsToRemove
+                    Dim hasUsedLocal = False
+                    For Each variable In candidate.Declarators
+                        If Not nodesToRemove.Contains(variable) Then
+                            hasUsedLocal = True
+                            Exit For
+                        End If
+                    Next
+
+                    If Not hasUsedLocal Then
+                        nodesToRemove.Add(candidate)
+                        For Each variable In candidate.Declarators
+                            nodesToRemove.Remove(variable)
+                        Next
+                    End If
+                Next
+            End Sub
+
+            Public Overrides Sub RemoveNode(editor As DocumentEditor, node As SyntaxNode)
+                editor.RemoveNode(node)
```

Comment:
For C# we seem to be retaining the right side of the assignment, but not for VB. Is this due to the difference in the scenarios that both compiler generate the diagnostics?

Reply:
'a=b=0' is not a double assignment in VB but is in C#. C# code is more complex to cover such situations and reduce 'a=b=0' to either 'a=0' or to 'b=0' if necessary. Also C# code uses C# specific SyntaxNodes. Therefore, it seems better to have a separate code path for VB.

---
In reply to: [152385291](https://github.com/dotnet/roslyn-analyzers/pull/1336#discussion_r152385291) [](ancestors = 152385291)